File: packages/shared/src/types/session.ts
/**
 * Session Management Types
 * Used for tracking and managing development sessions between Core and Client
 */

import { DeviceInfo } from "./device";

export interface Session {
  id: string;
  projectName: string;
  projectPath: string;
  deviceId: string;
  deviceInfo: DeviceInfo;
  createdAt: number;
  lastActivity: number;
  status: SessionStatus;
  connectionType: ConnectionType;
}

export enum SessionStatus {
  PENDING = 'pending',
  CONNECTED = 'connected',
  BUILDING = 'building',
  READY = 'ready',
  ERROR = 'error',
  DISCONNECTED = 'disconnected',
}

export enum ConnectionType {
  MINIMAL = 'minimal', // Manual APK installation
  FASTER = 'faster', // ADB over TCP
}

export interface SessionConfig {
  enableHotReload: boolean;
  enableLogs: boolean;
  autoReconnect: boolean;
  buildOnSave: boolean;
  port: number;
}

export interface SessionToken {
  sessionId: string;
  token: string;
  expiresAt: number;
  serverUrl: string;
  wsUrl: string;
}

export interface QRCodeData {
  sessionId: string;
  serverUrl: string;
  wsUrl: string;
  token: string;
  projectName: string;
  version: string;
}

File: packages/client/app/src/main/java/com/jetstart/client/data/models/Session.kt
package com.jetstart.client.data.models

data class Session(
    val id: String,
    val token: String,
    val serverUrl: String,
    val wsUrl: String,
    val projectName: String,
    val version: String
)

data class QRCodeData(
    val sessionId: String,
    val serverUrl: String,
    val wsUrl: String,
    val token: String,
    val projectName: String,
    val version: String
)